src/compiler/compile/render_ssr/handlers/Element.ts

```diff
@@ -176,11 +176,15 @@ export default function (node: Element, renderer: Renderer, options: RenderOptio
 	}
 
 	function add_close_tag() {
-		if (!is_void(node.name)) {
-			renderer.add_string('</');
-			add_tag_name();
-			renderer.add_string('>');
+		if (node.tag_expr.node.type === 'Literal') {
+			if (!is_void(node.tag_expr.node.value as string)) {
+				renderer.add_string('</');
+				add_tag_name();
+				renderer.add_string('>');
+			}
+			return;
 		}
+		renderer.add_expression(x`((tag) => @is_void(String(tag || '')) ? '' : \`</\${tag}>\`)(${node.tag_expr.node})`);
```

Comment:
i feel this there's potentially new problem with this, which is that, what if you have

```sv
<script>
  let tag = 'input';
</script>

<svelte:element this={tag}>
   hahaha
</svelte:element>
```

this will render 

```html
<input>
hahaha
```

which is wrong.

we need some sort of runtime validation to throw error when trying to render void elements with children element

----

@dummdidumm i wonder are we able to define `svelte:element` type definition as such that if `this` attribute belongs to a [void element](https://github.com/sveltejs/svelte/blob/master/src/compiler/utils/names.ts#L120-L124), it should not have any child elements

Reply:
I fear that's not possible. And I also don't think that it would help much, because the tag will be of type `string` 99% percent of the time, and then we couldn't check that anyway.

With regards to runtime validation: In dev mode I would be ok with that, although I don't know how many people will actually run into that.

Reply:
I added validation at runtime.
But still I didn't handle conditional situation like below.
Do we need to handle such case also?

```svelte
<script>
	let tag = "p";
</script>

<button on:click="{() => tag === 'p' ? tag = 'br' : tag = 'p'}">Click Me</button>
<svelte:element this="{tag}">
	{#if tag === 'p'}This is 'p'{/if}
</svelte:element>
```

Reply:
I would say no, this is too much of an edge case to justify the additional maintenance overhead.

Reply:
i think we can choose to warn over throwing error, but we should keep the behavior the same, meaning, always do not render children for void elements